addons/l10n_sa_edi/models/account_tax.py

```diff
@@ -40,3 +39,13 @@ def _l10n_sa_constrain_is_retention(self):
         for tax in self:
             if tax.amount >= 0 and tax.l10n_sa_is_retention and tax.type_tax_use == 'sale':
                 raise UserError(_("Cannot set a tax to Retention if the amount is greater than or equal 0"))
+
+    @api.depends('ubl_cii_tax_category_code')
+    def _compute_ubl_cii_requires_exemption_reason(self):
+        for tax in self:
+            tax.ubl_cii_requires_exemption_reason = (
+                tax.ubl_cii_tax_category_code in ['AE', 'E', 'G', 'O', 'K', 'Z'] and
+                tax.type_tax_use == 'sale' and
+                tax.country_code == 'SA' and
+                tax.amount != 0
+            )
```

Comment:
you do an override but could be an extend here ? Also do we want those new condition for every companies?